flask_oidc/__init__.py

```diff
@@ -187,11 +196,115 @@ def init_app(self, app):
         except KeyError:
             pass
 
-    def load_secrets(self, app):
-        # Load client_secrets.json to pre-initialize some configuration
-        return _json_loads(open(app.config['OIDC_CLIENT_SECRETS'],
-                                   'r').read())
-        
+    def init_provider(self, provider):
+        """
+        Do setup for a specific provider
+
+        :param provider: The provider to initialize.
+        :type provider: Dictionary with at lease 'base_url' item
+        """
+
+        secrets = self.load_secrets(provider)
+        assert secrets != None, "Problem with loading secrets"
```

Comment:
Note that `assert` statements get removed when a python optimized bytecode file is written, as happens in packaging on various linux distros.
Please turn this into an explicit check and `raise ValueError`.